code/data_structures/src/splay_tree/splay.cpp

```diff
@@ -0,0 +1,283 @@
+#include <iostream>
+#include <stdlib.h>
+
+struct Node
+{
+    int value;
+    Node *left;
+    Node *right;
+    Node *parent;
+};
+Node *g_root;
```

Comment:
Please avoid global variables. Even if you mark them so, a local one is still better.